spec.emu

```diff
@@ -209,6 +209,713 @@ contributors: Ujjwal Sharma, Younies Mahmoud
+
+    <emu-clause id="sec-temporal-totemporaldurationrecord" type="abstract operation">
+      <h1>
+        ToTemporalDurationRecord (
+          _temporalDurationLike_: an ECMAScript language value,
+        )
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It converts _temporalDurationLike_ to a Duration Record and returns it.</dd>
+      </dl>
+      <emu-alg>
+        1. If Type(_temporalDurationLike_) is not Object, then
+          1. Let _string_ be ? ToString(_temporalDurationLike_).
+          1. Return ? ParseTemporalDurationString(_string_).
+        1. If _temporalDurationLike_ has an [[InitializedTemporalDuration]] internal slot, then
+          1. Return ! CreateDurationRecord(_temporalDurationLike_.[[Years]], _temporalDurationLike_.[[Months]], _temporalDurationLike_.[[Weeks]], _temporalDurationLike_.[[Days]], _temporalDurationLike_.[[Hours]], _temporalDurationLike_.[[Minutes]], _temporalDurationLike_.[[Seconds]], _temporalDurationLike_.[[Milliseconds]], _temporalDurationLike_.[[Microseconds]], _temporalDurationLike_.[[Nanoseconds]]).
+        1. Let _result_ be a new Duration Record.
+        1. Let _any_ be *false*.
+        1. For each row of <emu-xref href="#table-duration-components"></emu-xref>, except the header row, in table order, do
+          1. Let _prop_ be the Property Name value of the current row.
+          1. Let _val_ be ? Get(_temporalDurationLike_, _prop_).
+          1. If _val_ is *undefined*, then
+            1. Set _result_'s field whose name is the Field Name value of the current row to 0.
+          1. Else,
+            1. Set _any_ to *true*.
+            1. Let _val_ be ? ToIntegerWithoutRounding(_val_).
+            1. Set _result_'s field whose name is the Field Name value of the current row to _val_.
+        1. If _any_ is *false*, then
+          1. Throw a *TypeError* exception.
+        1. If ! IsValidDuration(_result_.[[Years]], _result_.[[Months]], _result_.[[Weeks]] _result_.[[Days]], _result_.[[Hours]], _result_.[[Minutes]], _result_.[[Seconds]], _result_.[[Milliseconds]], _result_.[[Microseconds]], _result_.[[Nanoseconds]]) is *false*, then
+          1. Throw a *RangeError* exception.
+        1. Return _result_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-isvalidduration" type="abstract operation">
+      <h1>
+        IsValidDuration (
+          _years_: an integer,
+          _months_: an integer,
+          _weeks_: an integer,
+          _days_: an integer,
+          _hours_: an integer,
+          _minutes_: an integer,
+          _seconds_: an integer,
+          _milliseconds_: an integer,
+          _microseconds_: an integer,
+          _nanoseconds_: an integer,
+        )
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It returns *true* if its arguments form valid input from which to construct a `Temporal.Duration`, and *false* otherwise.</dd>
```

Comment:
there's no such thing as a `Temporal.Duration` until Temporal lands; this is pretty confusing.